etl/steps/data/garden/oecd/2024-04-10/income_distribution_database.py

```diff
@@ -0,0 +1,219 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import List
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+from structlog import get_logger
+from tabulate import tabulate
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Initialize logger.
+log = get_logger()
+
+# Define new names for categories
+INDICATOR_NAMES = {
+    "Gini (disposable income)": "gini_disposable",
+    "Gini (gross income)": "gini_gross",
+    "Gini (market income)": "gini_market",
+    "P50/P10 disposable income decile ratio": "p50_p10_ratio_disposable",
+    "P90/P10 disposable income decile ratio": "p90_p10_ratio_disposable",
+    "P90/P50 disposable income decile ratio": "p90_p50_ratio_disposable",
+    "Palma ratio (disposable income)": "palma_ratio_disposable",
+    "Quintile share ratio (disposable income)": "s80_s20_ratio_disposable",
+    "Poverty rate based on disposable income": "headcount_ratio_disposable",
+    "Poverty rate based on market income": "headcount_ratio_market",
+}
+
+POVERTY_LINES = {
+    "Not applicable": "Not applicable",
+    "50% of the national\xa0median disposable income": "50_median",
+    "60% of the national\xa0median disposable income": "60_median",
+}
+
+AGE_GROUPS = {"From 18 to 65 years": "Working population", "Over 65 years": "Over 65 years", "Total": "Total"}
+
+# Set table format when printing
+TABLEFMT = "pretty"
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("income_distribution_database")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["income_distribution_database"].reset_index()
+
+    #
+    # Process data.
+    tb = rename_and_create_columns(tb)
+
+    tb = create_relative_poverty_columns(tb)
+
+    tb = geo.harmonize_countries(
+        df=tb,
+        countries_file=paths.country_mapping_path,
+    )
+
+    tb = sanity_checks(tb)
+
+    tb = tb.format(["country", "year", "age"])
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def rename_and_create_columns(tb: Table) -> Table:
+    """
+    Rename categories in measure, poverty_line and age columns and make the table wide.
+    Also, add a gini_reduction column.
+    """
+    # Assert if all keys of dictionary are in the columns.
+    assert set(INDICATOR_NAMES.keys()) == set(
```

Comment:
You don't need `unique` here, it's enough to do `set(tb["measure"])`. Idem in the next lines.